google/resource_container_cluster.go

```diff
@@ -637,43 +657,49 @@ func resourceContainerClusterCreate(d *schema.ResourceData, meta interface{}) er
 }
 
 func resourceContainerClusterRead(d *schema.ResourceData, meta interface{}) error {
-	containerApiVersion := getContainerApiVersion(d, ContainerClusterBaseApiVersion, ContainerClusterVersionedFeatures)
+	containerAPIVersion := getContainerApiVersion(d, ContainerClusterBaseApiVersion, ContainerClusterVersionedFeatures)
 	config := meta.(*Config)
 
 	project, err := getProject(d, config)
 	if err != nil {
 		return err
 	}
 
-	zoneName, err := getZone(d, config)
-	if err != nil {
-		return err
+	var zoneName string
+	locations := []string{}
+	regionName, isRegionalCluster := d.GetOk("region")
+	if !isRegionalCluster {
+		zoneName, err = getZone(d, config)
+		if err != nil {
+			return err
+		}
+		locations = append(locations, zoneName)
 	}
 
 	cluster := &containerBeta.Cluster{}
+	var clust interface{}
```

Comment:
Did this work for you when you tested it? I've tried this approach (just doing the `Convert` from an interface at the end) in the past and it failed for me because Convert needed the parameter passed to be a struct of a certain type (so it can recurse through the fields).

Reply:
Well.. It compiled right and the tests passed. From that, I figured it works. Any other tests I can run to be doubly sure?

Reply:
@danawillow What did you to break this? I'd like to make sure this will not break later.

Reply:
Looks like it works! I must have done something slightly differently when I tried doing it this way in the past.